sklearn/feature_selection/tests/test_base.py

```diff
@@ -1,12 +1,15 @@
 import numpy as np
 from scipy import sparse as sp
+from scipy.stats import pearsonr, spearmanr
```

Comment:
I guess the signature of `call_scorer` needs to be changed then. How about

```
def call_scorer(X, y=None, **kwargs)
```

My guess is that you're getting that `TypeError` on the line

```
score_func_ret = score_func(X[:, i], y, **kwargs)
```

and that's because `y` is being interpreted as the second keyword argument to `np.var`, which is `axis`. A decision needs to be made about how you're going to handle cases when the scoring function takes only `X`, but not a `y`.